cmd/geth/snapshot.go

```diff
@@ -149,6 +166,76 @@ It's also usable without snapshot enabled.
 	}
 )
 
+func accessDb(ctx *cli.Context, stack *node.Node) (ethdb.Database, error) {
+	chaindb := utils.MakeChainDatabase(ctx, stack, false, true)
+	defer chaindb.Close()
+	headBlock := rawdb.ReadHeadBlock(chaindb)
+	if headBlock == nil {
+		return nil, errors.New("failed to load head block")
+	}
+	//Make sure the MPT and snapshot matches before pruning, otherwise the node can not start.
+	_, err := snapshot.New(chaindb, trie.NewDatabase(chaindb), 256, int(ctx.GlobalUint64(utils.TriesInMemoryFlag.Name)), headBlock.Root(), false, false, false)
+	if err != nil {
+		return nil, err // The relevant snapshot(s) might not exist
+	}
+	return chaindb, nil
+}
+
+func pruneBlock(ctx *cli.Context) error {
+	stack, config := makeConfigNode(ctx)
+	defer stack.Close()
+	chaindb, err := accessDb(ctx, stack)
+	if err != nil {
+		utils.Fatalf("MPT and snapshot sanity check failed: %v", err)
+		return err
+	}
+	var oldAncientPath, newAncientPath string
+	path, _ := filepath.Split(ctx.GlobalString(utils.AncientFlag.Name))
+	if path != "" {
+		if !filepath.IsAbs(path) {
+			path = stack.ResolvePath(path)
+		}
+		oldAncientPath = path + "ancient"
+		newAncientPath = path + "ancient_back"
+	} else {
+		utils.Fatalf("Prune failed, did not specify the AncientPath")
+		return errors.New("Prune failed, did not specify the AncientPath")
```

Comment:
`Fatalf` will do `os.Exit(1)`, please just return, do not `Fatalf`

Reply:
Done